pkg/remediator/oldpoddeleter.go

```diff
@@ -0,0 +1,86 @@
+package remediator
+
+import (
+	"context"
+	"github.com/aksgithub/kube_remediator/pkg/k8s"
+	"go.uber.org/zap"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sync"
+	"time"
+)
+
+type OldPodDeleter struct {
+	client          k8s.ClientInterface
+	logger          *zap.Logger
+}
+
+// Entrypoint
+func (p *OldPodDeleter) Run(ctx context.Context, wg *sync.WaitGroup) {
+	defer waitDone(wg)
+
+	ticker := time.NewTicker(1 * time.Hour)
+	defer ticker.Stop()
+
+	p.logger.Info("Starting")
+
+	// Run on start
+	p.deleteOldPods()
+
+	for {
+		select {
+			case <-ticker.C:
+				p.deleteOldPods()
+			case <-ctx.Done():
+				p.logger.Info("Stopping", zap.String("reason", "Signal"))
+				return
+			}
+       }
+}
+
+func (p *OldPodDeleter) deleteOldPods() {
+	p.logger.Info("Running")
+
+	// get all pods that opted in to deletion
+	pods, err := p.client.GetPods("", metav1.ListOptions{
+		LabelSelector: "kube-remediator/OldPodDeleter=true",
+	})
+	if err != nil {
+		p.logger.Error("Error getting pod list: ", zap.Error(err))
+		return
+	}
+
+	// deleteOldPods those that are too old
+	cutoff := time.Now().Add(-24 * time.Hour)
```

Comment:
what if definition of "old" varies among applications? Debugging session more than 24 hours might be unrealistic but just a thought.

Reply:
how about read cutoff from application pod's annotation?